komodo-relational-commands/src/main/java/org/komodo/relational/commands/server/ServerManager.java

```diff
@@ -0,0 +1,219 @@
+        this.wsStatus = wsStatus;
+    }
+    
+    /**
+     * Get the server defined for this session.  If the default server does not exist, create it.
+     * @return the Teiid server
+     * @throws KException if error occurs
+     */
+    public Teiid getDefaultServer( ) throws KException {
+        if( this.server == null ) {
+            UnitOfWork uow = this.wsStatus.getTransaction();
+            KomodoObject serversNode = getServersNode( );
+
+            KomodoObject serverObj = null;
+            if(!serversNode.hasChild(uow, DEFAULT_SERVER_NAME)) {
+                serverObj = serversNode.addChild( uow, DEFAULT_SERVER_NAME, KomodoLexicon.Teiid.NODE_TYPE);
+            } else {
+                serverObj = serversNode.getChild( uow, DEFAULT_SERVER_NAME, KomodoLexicon.Teiid.NODE_TYPE );
+            }
+            if( serverObj!=null && Teiid.RESOLVER.resolvable( uow, serverObj ) ) {
+                this.server = Teiid.RESOLVER.resolve( uow, serverObj );
+            }
+        }
+        return this.server;
+    }
+    
+    /**
+     * Clears the default server.  This can be used when the repo object properties change - forces the default server to be refreshed.
+     */
+    public void clearDefaultServer() {
+        this.server = null;
+    }
+    
+    /**
+     * Requests connection to default server
+     * @return <code>true</code> if the server is connected, <code>false</code> if not.
+     * @throws KException if error occurs
+     */
+    public boolean connectDefaultServer( ) throws KException {
+        if(!isDefaultServerConnected( )) {
+            TeiidInstance defaultTeiid = getDefaultTeiidInstance( );
+            if(defaultTeiid!=null) {
+                try {
+                    defaultTeiid.connect();
+                } catch (Exception ex) {
+                    throw new KException(ex);
+                }
+                this.isConnected = defaultTeiid.isConnected();
+            }
+        }
+        return this.isConnected;
+    }
+
+    /**
+     * Requests default server disconnect
+     * @return <code>true</code> if the server is connected, <code>false</code> if not.
+     * @throws KException if error occurs
+     */
+    public boolean disconnectDefaultServer( ) throws KException {
+        if(isDefaultServerConnected( )) {
+            TeiidInstance defaultTeiid = getDefaultTeiidInstance( );
+            if(defaultTeiid!=null) {
+                defaultTeiid.disconnect();
+                this.isConnected = defaultTeiid.isConnected();
+            }
+        }
+        clearDefaultServer();
+        return this.isConnected;
+    }
+    
+    /**
+     * Determine if the default server is connected.
+     * @return <code>true</code> if the server is connected, <code>false</code> if not.
+     * @throws KException if error occurs
+     */
+    public boolean isDefaultServerConnected( ) throws KException {
+        UnitOfWork uow = this.wsStatus.getTransaction();
```

Comment:
Cannot short circuit as currently written.  getDefaultServer will create the server if it doesn't exist...